packages/liveblocks-core/src/types/QueryMetadata.ts

```diff
@@ -0,0 +1,20 @@
+import type { BaseMetadata } from "./BaseMetadata";
+
+type QueryMetadataStringValue<T extends string> =
+  | T
+  | {
+      operator: "startsWith";
+      value: T;
+    };
+
+/**
+ * This type can be used to build a metadata query string (compatible
+ * with `@liveblocks/query-parser`) through a type-safe API.
+ *
+ * In addition to exact values (`:` in query string), it adds:
+ * - to strings:
+ *  - `startsWith` (`^` in query string)
```

Comment:
Is there a way to make this only available if the value is a `string` type?

`boolean` and `number` values can only have an exact match `:`

Reply:
Yep this type only adds `startsWith` to `string` values, all other values remain exact-only.

Reply:
Great! Sorry I think I misread the comment 😅 

Thanks